joomla/joomla-cms
# Proper handling of form validation messages

In case of a validation failure, the (error) message is not displayed. This is due to the fact that an Exception object is pushed onto the JForm::errors array, while this must be a text string apparently.

Line 1212
array_push($this->errors, $valid);

has now been changed into
array_push($this->errors, $valid->getMessage());

Which seems to solve the problem.

Test instructions: 

This is for testing that a custom rule error message is not shown by Joomla in a component configuration.

Test instructions:
- Download test component from [com_formtest](http://csvimproved.com/com_formtest.zip)
- Install com_formtest
- Open com_formtest in administrator
- Click on Options
- Fill in the letters ab
- Click on Save
- The problem: no error message is shown
- Apply patch
- Fill in the letters ab
- Click on Save
- Problem solved: an error message is shown
- Fill in the number 2
- Click on Save
- Success message is shown
